### PR TITLE
Add nightly product functional testing

### LINKED ISSUE #652: Add nightly product functional testing
# 💡 Summary #

What is the work, as a high-level summary?

Create a GitHub Action workflow that runs all of the product testing on a nightly basis.

## Motivation and context ##

Why does this work belong in this project?

Currently this testing is run on an ad hoc basis.  Running it nightly will ensure that testing constantly reveals product issues that need to be resolved.

This would be useful because...

Continuous testing helps issues to be found and resolved sooner.

## Implementation notes ##

Please provide details for implementation, such as:

GitHub Action workflow
Calls Products.Tests.ps1

There are plenty other ways to improve this workflow, but those can be future user stories.

## Acceptance criteria ##

How do we know when this work is done?

A workflow runs every weeknight that tests the various products.

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

Create an initial pipeline that conducts functional testing for one product every weekday night.

## 💭 Motivation and context ##

These tests ensure that ScubaGear continues to effectively test positive and negative settings for Sharepoint.  As functional testing can take a long time, they are running nightly to avoid slowly down work.

Closes #652 

## 🧪 Testing ##

Created new workflow.
Created script that calls the product testing script.
Created a PSON file that provides the input parameters for testing.  This provides a data-driven approach.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues (see #731).

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: .github/workflows/run_functional_product_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      env:
+        PfxBase64: ${{ secrets.NIGHTLY_TEST_BUILD_PFX }}
+    - name: Import PFX
+      run: Import-PfxCertificate -Password (ConvertTo-SecureString -String $env:PfxPassword -AsPlainText -Force) -CertStoreLocation Cert:\CurrentUser\My -FilePath ./key.pfx
+      env:
+        PfxPassword: ${{ secrets.NIGHTLY_TEST_BUILD_PW }}
+    - name: Setup for Scuba
+      run: ./SetUp.ps1
+    - name: Install Selenium
+      run: Install-Module -Name Selenium -RequiredVersion 3.0.1
```
Comment (crutchfield) [inline .github/workflows/run_functional_product_tests.yaml:39]: Not bad but why restrict to version 3.0.1 instead of just latest?

File: Testing/Functional/Products/SetupFunctionalProductTesting.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Credentials
+
+Two things are needed in order to authenticate to the various services that are being tested.  The first is a personal certificate.  Ask Nanda for the documentation on how to create a ScubaGear personal certificate.  Follow the instructions and note the value of the thumbprint.
+
+The second is to add your certificate to the service principal.  Send the certificate and the thumbprint to Nanda, and he will add it to the service principal and provide the id (aka the App ID).
+
+## Powershell 5.1
+
+ScubaGear runs on an outdated version of Powershell, version 5.1.  To run this version in Windows 10, search for "powershell" (not "pwsh"), and run it as administrator.  Run all commands in this shell.
```
Comment (crutchfield) [inline Testing/Functional/Products/SetupFunctionalProductTesting.md:17]:
```suggestion
ScubaGear runs on PowerShell, version 5.1.  To run this version in Windows 10, search for "powershell" (not "pwsh"), and run it as administrator.  Run all commands in this shell.
```

File: Testing/Functional/Products/Tests/CallProductTests.ps1 [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# James Garriss
+# Dec 2023
+# The purpose of this script is to run the functional tests for all the products.
+# NOTE: At the moment it only does this for one product, Sharepoint.  Others
+#       will be added.
+# To run this script, call it from the root of the repo, like so:
+# ./Testing/Functional/Products/Tests/CallProductTests.ps1
+
+# Setup directories
+$testDataFile = 'Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson'
```
Comment (crutchfield) [inline Testing/Functional/Products/Tests/CallProductTests.ps1:10]: Interesting.  I would have expected this to be a parameter that is passed into this script (from user or workflow).

File: Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@{
+    Thumbprint = 'FD6219CFBF937830616E43DF9C81BCF7D22FE03F'
```
Comment (crutchfield) [inline Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson:2]: Did not expect hard coded thumbprint.  You can get from certificate.

File: Testing/Functional/Products/Tests/CallProductTests.ps1
Comment (crutchfield) [inline Testing/Functional/Products/Tests/CallProductTests.ps1:1]: Need to refactor to be able to run manually with my local certificate or GitHub secret provided secret.

**Review by crutchfield — CHANGES_REQUESTED** [review]
Requesting a couple refactoring items.

File: .github/workflows/run_functional_product_tests.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Author: James Garriss
+# Purpose: Run the product functional tests every night
+
+name: Nightly Product Functional Tests
+
+# Run this workflow at midnight 
+# on every Mon to Fri (b/c GMT -5)
+on:
+  schedule:
+  - cron: '0 4 * * 1-5'
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml:10]: Done.

File: .github/workflows/run_functional_product_tests.yaml (hunk 1 above)
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml]: done

File: .github/workflows/run_functional_product_tests.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  workflow_dispatch: 
+  push:
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml:12]: You are correct.  I only put that there for testing, and I forgot to remove it.  I will do so.

File: Testing/Functional/Products/SetupFunctionalProductTesting.md (hunk 2 above)
Comment (james-garriss) [reply Testing/Functional/Products/SetupFunctionalProductTesting.md:17]: Done.

File: Testing/Functional/Products/Tests/CallProductTests.ps1 (hunk 3 above)
Comment (james-garriss) [reply Testing/Functional/Products/Tests/CallProductTests.ps1:10]: If you read the TODO comment in the script, you will see that this is hard-coded for the first version.  The next US (731) will add tests for additional products.  As it says, the plan is to read all the PSON files in the directory and test them all.  We have here a to-MAY-to vs to-MAH-to thing.  Do you want the workflow to have to keep track of all possible tests and call them one by one?  Or do you want the script to find all PSON files on its own and run those tests?  I think I prefer the later, but neither is really wrong.  If you strong feel like it should do the former, please create a new conversation, and I will re-write it accordingly.

File: Testing/Functional/Products/Tests/CallProductTests.ps1
Comment (james-garriss) [reply Testing/Functional/Products/Tests/CallProductTests.ps1:1]: I'm going to disagree with you here.  The purpose of this script is not to provide general purpose code for running the tests.  You have already written that with Products.Tests.ps1.  The purpose of this script is to enable the nightly testing workflow to use Products.Tests.ps1.  (This is back to the question of where does the workflow code go, in the workflow file or in an external file.)  If you want to test with your local cert, you can already do that (and you already have) via Products.Tests.ps1.

Now perhaps I have chosen a poor location for this file.  Maybe, because it is workflow code, it should be moved into the workflow file itself or into the .github/workflow folder.  I would be happy to change location if that is necessary to make its purpose clearer.

Comment (james-garriss) [reply Testing/Functional/Products/Tests/CallProductTests.ps1:1]: Also clarified the purpose of this script in its comments.

File: .github/workflows/run_functional_product_tests.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (crutchfield) [reply .github/workflows/run_functional_product_tests.yaml:12]: One thing I did that seems to work out well when working on workflows is to add the workflow yaml to the path for a push.  Does nto cover all cases but helps.  If I have a main supporting PowerShell script I add it to path as well.

[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml:12]: Removed.

File: Testing/Functional/Products/Tests/CallProductTests.ps1
Comment (crutchfield) [reply Testing/Functional/Products/Tests/CallProductTests.ps1:1]: That seems valid rationale.

File: Testing/Functional/Products/SetupFunctionalProductTesting.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+In order to run the functional tests, the following items need to be setup.
+
+## Windows
+
+This testing needs to be run on a Windows computer or VM.
+
+## Credentials
+
+Two things are needed in order to authenticate to the various services that are being tested.  The first is a personal certificate.  Ask Nanda for the documentation on how to create a ScubaGear personal certificate.  Follow the instructions and note the value of the thumbprint.
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply Testing/Functional/Products/SetupFunctionalProductTesting.md:11]: Done.

File: .github/workflows/run_functional_product_tests.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+      uses: actions/checkout@v4
+    - name: Create PFX
+      run: |
+        New-Item -ItemType file -Path ./key.txt
+        Set-Content -Path ./key.txt -Value $env:PfxBase64
+        certutil -decode ./key.txt ./key.pfx
+      env:
+        PfxBase64: ${{ secrets.NIGHTLY_TEST_BUILD_PFX }}
+    - name: Import PFX
+      run: Import-PfxCertificate -Password (ConvertTo-SecureString -String $env:PfxPassword -AsPlainText -Force) -CertStoreLocation Cert:\CurrentUser\My -FilePath ./key.pfx
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml]: Tested.  Works.  Done.

File: Testing/Functional/Products/SetupFunctionalProductTesting.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+## Powershell 5.1
+
+ScubaGear runs on Powershell 5.1.  To run this version in Windows 10, search for "powershell" (not "pwsh"), and run it as administrator.  Run all commands in this shell.
+
+## Pester
+
+"Pester is a testing and mocking framework for PowerShell." ([Pester Quick Start](https://pester.dev/docs/quick-start))
+
+Install Pester by running this command as administrator:
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply Testing/Functional/Products/SetupFunctionalProductTesting.md:23]: Done.

File: Testing/Functional/Products/Tests/CallProductTests.ps1 [hunk 10]
```diff
@@ -0,0 +1,32 @@
+# James Garriss
+# Dec 2023
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply Testing/Functional/Products/Tests/CallProductTests.ps1:2]: Done.

File: Testing/Functional/Products/Tests/Cleanup.ps1
[the remark this answers is not in the document]
Comment (james-garriss) [reply Testing/Functional/Products/Tests/Cleanup.ps1:1]: Done.

File: Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson
[the remark this answers is not in the document]
Comment (james-garriss) [reply Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson:1]: @schrolla, Ok.  If you will add the contents of this PSON file to a secret and provide me with the secret name, please, then I will read the secret and pass it to the script.

[the remark this answers is not in the document]
Comment (james-garriss) [reply Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson:1]: I think this version (for this US) will be just a hashtable (exactly what is in the PSON), and the next version (for next US) will an array of hashtables, one for each product/tenant that is tested.

File: .github/workflows/run_functional_product_tests.yaml [hunk 11]
```diff
@@ -0,0 +1,56 @@
+# Author: James Garriss
+# Purpose: Run the product functional tests every night
+
+name: Nightly Product Functional Tests
+
+# Run this workflow at midnight 
+# on every Sun to Thr (b/c GMT -5)
+on:
+  push:
```
Comment (crutchfield) [inline .github/workflows/run_functional_product_tests.yaml:9]: I assume we do not want this to trigger on push.  Is there a need for this?

File: .github/workflows/run_functional_product_tests.yaml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      run: ./SetUp.ps1
+    - name: Install Selenium
+      run: Install-Module -Name Selenium
+    - name: Update Selenium
+      run: ./Testing/Functional/SmokeTest/UpdateSelenium.ps1
+    - name: Test SharePoint
+      run: |
+        # Read thumbprint from previous job.
+        $thumbprint = "${{ steps.thumbprint.outputs.THUMBPRINT }}"
+        ./Testing/Functional/Products/Tests/CallProductTests.ps1 $thumbprint
```
Comment (crutchfield) [inline .github/workflows/run_functional_product_tests.yaml:56]: Not very intuitive that CallProductTests will test "just" SharePoint.  What is the long term intent here?

File: .github/workflows/run_linter_powershell.yaml
Comment (crutchfield) [inline .github/workflows/run_linter_powershell.yaml:1]: What does this have to do with nightly product functional testing?

File: .github/workflows/run_linter_yaml.yaml
Comment (crutchfield) [inline .github/workflows/run_linter_yaml.yaml:1]: Related to PR?

File: .github/workflows/run_opa_tests.yaml [hunk 13]
```diff
@@ -26,7 +26,7 @@ jobs:
     - name: Setup OPA
       uses: open-policy-agent/setup-opa@v2
       with:
-        version: <0.50
+        version: latest
```
Comment (crutchfield) [inline .github/workflows/run_opa_tests.yaml:29]: We might consider using our bespoke scripts since there is/will be an expectation of which version and where the OPA agent is placed.

File: .github/workflows/run_opa_tests.yaml
Comment (crutchfield) [inline .github/workflows/run_opa_tests.yaml:1]: Seems out of scope of PR - but as an observation, we are seperating linting and testing in PowerShell but are combining them here.  I don't know this causes any concern but is inconsistent.

File: SetUp.ps1 [hunk 14]
```diff
@@ -36,7 +36,7 @@ param(
     [Parameter(Mandatory = $false, HelpMessage = 'The version of OPA Rego to be downloaded, must be in "x.x.x" format')]
     [Alias('version')]
     [string]
```
Comment (crutchfield) [inline SetUp.ps1:38]:
```suggestion
    [version]
```

File: Testing/Functional/Products/SetupFunctionalProductTesting.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
+# Setup Testing
+
+In order to run the functional tests, the following items need to be setup.
+
+## Windows
+
+This testing needs to be run on a Windows computer or VM.
+
+## Credentials
```
Comment (crutchfield) [inline Testing/Functional/Products/SetupFunctionalProductTesting.md:9]: Functional testing can use certificates or manual login.

File: Testing/Functional/Products/Tests/CallProductTests.ps1 [hunk 16]
```diff
@@ -0,0 +1,30 @@
+# The purpose of this script is to enable a GitHub Action workflow to
+# run the functional tests for all the products.
+# NOTE: At the moment it only does this for one product, Sharepoint.  Others
+#       will be added.
+# To run this script, call it from the root of the repo, like so:
+# ./Testing/Functional/Products/Tests/CallProductTests.ps1 <value of thumbprint>
+
+# The thumbprint of the cert used to access the product.
+$thumbprint = $args[0]
```
Comment (crutchfield) [inline Testing/Functional/Products/Tests/CallProductTests.ps1:9]: Declare as a parameter to the script.

File: Testing/Functional/Products/Tests/RemoveOutputFolders.ps1 [hunk 17]
```diff
@@ -0,0 +1,16 @@
+# The purpose of this script is to delete all the temp folders created by running ScubaGear.
+
+$thisDir = Get-Location
```
Comment (crutchfield) [inline Testing/Functional/Products/Tests/RemoveOutputFolders.ps1:3]: Declaring the target directory as a parameter and defaulting to current location may make this more reusable.

File: Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson [hunk 18]
```diff
@@ -0,0 +1,8 @@
+@{
+    TenantDomain = 'y2zj1.onmicrosoft.com'
+    TenantDisplayName = 'y2zj1'
+    AppId = 'db791fde-1ff0-493f-be3d-03d9ba6cb087'
```
Comment (crutchfield) [inline Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson:4]: It is my opinion that this info should be in GitHub as a secret similar to Smoke Test.

**Review by crutchfield — CHANGES_REQUESTED** [review]
Seems to include files and changes not related to the PR.  Maybe result of merge?  Anyway looks good for most part.  Review my comments and adjudicate.

File: Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson
[the remark this answers is not in the document]
Comment (james-garriss) [reply Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson:1]: Nanda put the PSON into a secret, and I am now using that.  The PSON file is no longer in the repo.

File: .github/workflows/run_functional_product_tests.yaml (hunk 12 above)
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml]: The comments in this script already address this:

```
The purpose of this script is to enable a GitHub Action workflow to
run the functional tests for all the products.
NOTE: At the moment it only does this for one product, Sharepoint.  Others
      will be added.
```

We are being agile, staring with one product and adding other later.  Issue 731, for example, will add Teams.

File: SetUp.ps1 (hunk 14 above)
Comment (james-garriss) [reply SetUp.ps1:38]: I am not modifying that workflow. Please open a new issue and let whoever is in charge of that workflow know that it should be changed.

File: .github/workflows/run_functional_product_tests.yaml (hunk 11 above)
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml:8]: You are correct.  This was added back in order to make some changes requested by Addam.  It will be removed.

File: .github/workflows/run_linter_powershell.yaml
Comment (james-garriss) [reply .github/workflows/run_linter_powershell.yaml:1]: Nothing.  I had put my name as the author of the script, and Addam commented that we don't do that, so I removed my name from this script while I was editing the other.  Target of opportunity.  Didn't seem worth opening a whole new issue to do that, but maybe I should have?

File: .github/workflows/run_linter_yaml.yaml
Comment (james-garriss) [reply .github/workflows/run_linter_yaml.yaml:1]: Nothing.  I had put my name as the author of the script, and Addam commented that we don't do that, so I removed my name from this script while I was editing the other.  Target of opportunity.  Didn't seem worth opening a whole new issue to do that, but maybe I should have?

File: Testing/Functional/Products/Tests/CallProductTests.ps1 (hunk 16 above)
Comment (james-garriss) [reply Testing/Functional/Products/Tests/CallProductTests.ps1]: I declared both parameters in a `param`.  Done.

File: Testing/Functional/Products/Tests/RemoveOutputFolders.ps1 (hunk 17 above)
Comment (james-garriss) [reply Testing/Functional/Products/Tests/RemoveOutputFolders.ps1:3]: OBE.  This variable disappeared after moving the hashtable into a secret.  Done.

File: Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson (hunk 18 above)
Comment (james-garriss) [reply Testing/Functional/Products/Tests/TestData/sharepoint-commercial-data.pson]: Already done, thank you.

File: .github/workflows/run_functional_product_tests.yaml [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+jobs:
+  sharepoint-test:
+    name: SharePoint Test
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: powershell 
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v4
```
Comment (crutchfield) [inline .github/workflows/run_functional_product_tests.yaml:21]: Steps takes an array of object - needs aditinal indentation 
```suggestion
      - name: Checkout Repo
        uses: actions/checkout@v4
```

File: Testing/Functional/Products/Tests/CallProductTests.ps1 [hunk 20]
```diff
@@ -0,0 +1,31 @@
+# The purpose of this script is to enable a GitHub Action workflow to
+# run the functional tests for all the products.
+# NOTE: At the moment it only does this for one product, Sharepoint.  Others
+#       will be added.
+# To run this script, call it from the root of the repo, like so:
+# ./Testing/Functional/Products/Tests/CallProductTests.ps1 <params> <thumbprint>
```
Comment (crutchfield) [inline Testing/Functional/Products/Tests/CallProductTests.ps1:6]: Use PowerShell built in Comment Based Help so user can Get-Help on file usage: 
Get-Help .\Testing\Functional\Products\Tests\CallProductTests.ps1
 
```suggestion
<#
    .SYNOPSIS
    The purpose of this script is to enable a GitHub Action workflow to run the functional tests for all the products.
    .NOTE: 
    At the moment it only does this for one product, Sharepoint.  Others will be added.
    .EXAMPLE 
    To run this script, call it from the root of the repo, like so:
    ./Testing/Functional/Products/Tests/CallProductTests.ps1 <params> <thumbprint>
#>
```

File: Testing/Functional/Products/Tests/RemoveOutputFolders.ps1 (hunk 17 above)
Comment (crutchfield) [reply Testing/Functional/Products/Tests/RemoveOutputFolders.ps1:3]: I do not understand the resolved comment.  Variable still appears to be in baseline.

**Review by crutchfield — CHANGES_REQUESTED** [review]
Needs YAML structural changes to distinguish arrays from dictionaries.

File: .github/workflows/run_functional_product_tests.yaml [hunk 21]
```diff
@@ -0,0 +1,60 @@
+# Purpose: Run the product functional tests every night
+
+name: Nightly Product Functional Tests
+
+# Run this workflow at midnight 
+# on every Sun to Thr (b/c GMT -5)
+on:
+  schedule:
+  - cron: '0 4 * * 0-4'
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml:9]: Done.

File: .github/workflows/run_functional_product_tests.yaml (hunk 19 above)
Comment (james-garriss) [reply .github/workflows/run_functional_product_tests.yaml:21]: Done.

File: Testing/Functional/Products/Tests/CallProductTests.ps1 (hunk 20 above)
Comment (james-garriss) [reply Testing/Functional/Products/Tests/CallProductTests.ps1:6]: Done.

**Review by crutchfield — CHANGES_REQUESTED** [review]
See comments

File: Testing/Functional/Products/Tests/RemoveOutputFolders.ps1 (hunk 17 above)
Comment (james-garriss) [reply Testing/Functional/Products/Tests/RemoveOutputFolders.ps1:6]: I see what happened.  I had used this command in 2 different scripts, and I removed it from the other script.  Sorry, I got confused.

As far as changing this script to use an input parameter, no, I don't want to do that.  Eschew scope creep.  This script has a very specific purpose, it works great for that purpose.  I don't need it to be a more general-purpose script.  Similar to this discussion that we had for the other script, if you want it to be changed into something else, then please create a new issue and we can see if that's where the team wants to go.

File: Testing/Functional/Products/Tests/CallProductTests.ps1 [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+    .SYNOPSIS
+    The purpose of this script is to enable a GitHub Action workflow to run the functional tests for all the products.
+    .EXAMPLE
+    To run this script, call it from the root of the repo, like so:
+    ./Testing/Functional/Products/Tests/CallProductTests.ps1 <params> <thumbprint>
+#>
+
+# NOTE:
+# At the moment this script is only used for one product, Sharepoint.  Others
+# will be added over time.
```
Comment (crutchfield) [inline Testing/Functional/Products/Tests/CallProductTests.ps1:11]:
```suggestion
<#
    .SYNOPSIS
    The purpose of this script is to enable a GitHub Action workflow to run the functional tests for all the products.
    .EXAMPLE
    To run this script, call it from the root of the repo, like so:
    ./Testing/Functional/Products/Tests/CallProductTests.ps1 <params> <thumbprint>
    .NOTE
    At the moment this script is only used for one product, Sharepoint.  Others will be added over time.
#>
```

File: Testing/Functional/Products/Tests/RemoveOutputFolders.ps1 (hunk 17 above)
Comment (crutchfield) [reply Testing/Functional/Products/Tests/RemoveOutputFolders.ps1:6]: I am okay with defering changing.  We are currently modifying Scuba to be deployed from a package repo as a module. Part of this will modify where the reports are stored.  I expect the Functional Tests will be updated as well setup Scuba using the package repo - so the clean up here will need to accomodate (in this release).

File: Testing/Unit/Rego/Sharepoint/SharepointConfig_01_test.rego
Comment (crutchfield) [inline Testing/Unit/Rego/Sharepoint/SharepointConfig_01_test.rego:1]: Fails automated Run-OPA-Tests.
Run opa check Rego Testing/Unit/Rego --strict
  opa check Rego Testing/Unit/Rego --strict
  shell: /usr/bin/bash -e {0}
4 errors occurred:
Testing/Unit/Rego/Sharepoint/SharepointConfig_01_test.rego:[2](https://github.com/cisagov/ScubaGear/actions/runs/7397959623/job/20126059393?pr=733#step:4:2)6: rego_type_error: undefined function ReportDetailsBoolean
Testing/Unit/Rego/Sharepoint/SharepointConfig_01_test.rego:28: rego_type_error: undefined function ReportDetailsBoolean

File: Testing/Unit/Rego/Sharepoint/SharepointConfig_04_test.rego
Comment (crutchfield) [inline Testing/Unit/Rego/Sharepoint/SharepointConfig_04_test.rego:1]: Fails automated Run-OPA-Tests.
Run opa check Rego Testing/Unit/Rego --strict
  opa check Rego Testing/Unit/Rego --strict
  shell: /usr/bin/bash -e {0}
4 errors occurred:

Testing/Unit/Rego/Sharepoint/SharepointConfig_0[4](https://github.com/cisagov/ScubaGear/actions/runs/7397959623/job/20126059393?pr=733#step:4:5)_test.rego:2[6](https://github.com/cisagov/ScubaGear/actions/runs/7397959623/job/20126059393?pr=733#step:4:7): rego_type_error: undefined function ReportDetailsBoolean
Testing/Unit/Rego/Sharepoint/SharepointConfig_04_test.rego:2[8](https://github.com/cisagov/ScubaGear/actions/runs/7397959623/job/20126059393?pr=733#step:4:9): rego_type_error: undefined function ReportDetailsBoolean

**Review by crutchfield — CHANGES_REQUESTED** [review]
See comments
